spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMigrateTableProcedure.java

```diff
@@ -106,20 +108,41 @@ public void testMigrateWithOptions() throws IOException {
     sql("DROP TABLE IF EXISTS  %s", tableName + "_BACKUP_");
   }
 
-  @Test
-  public void testMigrateWithDropBackup() throws IOException {
+  private void testMigrateWithDropBackup(@Nullable String backupSuffix) throws IOException {
```

Comment:
nit: I am not really familiar with the usage of the `@Nullable` annotation. But how about directly passing the optional to the method as an argument. i.e couldn't you just pass `testMigrateWithDropBackup(Optional.ofNullable("_tmp"));` & `testMigrateWithDropBackup(Optional.ofNullable(null));` rather than constructing the optional inside the method. 
I am not familiar with this pattern you are using, but if it is some sort of standard coding practice please let me know.